src/hooks/useBanner.ts

```diff
@@ -0,0 +1,181 @@
+import { useMemo } from 'react';
+import dayjs from 'dayjs';
+import timezone from 'dayjs/plugin/timezone';
+import utc from 'dayjs/plugin/utc';
+import yaml from 'js-yaml';
+import * as yup from 'yup';
+import { BANNER_CONTENT_FILE, BANNER_NAMEPSACE } from '~/consts/banner';
+import { useK8sWatchResource } from '~/k8s';
+import { ConfigMap } from '~/types/configmap';
+import { bannerConfigYupSchema } from '~/utils/validation-utils';
+import { ConfigMapGroupVersionKind, ConfigMapModel } from './../models/config-map';
+
+dayjs.extend(utc);
+dayjs.extend(timezone);
+
+export type BannerType = 'info' | 'warning' | 'danger';
+export type BannerConfig = {
+  summary: string;
+  type: BannerType;
+  year?: string;
+  month?: string;
+  startTime?: string;
+  endTime?: string;
+  timeZone?: string;
+  dayOfWeek?: number; // 0-6 for Sunday-Saturday
+  dayOfMonth?: number; // 1-31 for the day of the month
+};
+
+export function inferRepeatType(banner: BannerConfig): 'weekly' | 'monthly' | 'none' {
+  // Sunday is 0, so we can not enjoy 'true' check here.
+  if (banner.dayOfWeek !== undefined) return 'weekly';
+  if (banner.dayOfMonth) {
+    if (banner.month) return 'none'; // specific date
+    return 'monthly'; // repeats every month on this day
+  }
+  return 'none'; // fallback
+}
+
+function convertToTimeZone(date: Date, timeZone: string): Date {
+  return dayjs(date).tz(timeZone).toDate();
+}
+
+// Fore repeated banners
+function parseHM(hm: string): number {
+  const [h, m] = hm.split(':').map(Number);
+  return h * 60 + m;
+}
+
+/**
+ * Parses and validates a YAML banner list string.
+ * Filters out any invalid entries silently.
+ */
+export const parseBannerList = (yamlContent: string): BannerConfig[] => {
+  try {
+    const parsed = yaml.load(yamlContent);
+
+    if (!Array.isArray(parsed)) return [];
+
+    const validBanners: BannerConfig[] = [];
+
+    for (const banner of parsed) {
+      try {
+        // Validate using your existing schema
+        const validated = bannerConfigYupSchema.validateSync(banner, {
+          strict: false,
+          abortEarly: false,
+        });
+        validBanners.push(validated as BannerConfig);
+      } catch (err) {
+        // eslint-disable-next-line no-console
+        console.warn('Invalid banner skipped:', banner);
+
+        if (err instanceof yup.ValidationError) {
+          // eslint-disable-next-line no-console
+          console.warn('Validation errors:', err.errors);
+        } else {
+          // eslint-disable-next-line no-console
+          console.warn('Unexpected validation error:', err);
+        }
+      }
+    }
+
+    return validBanners;
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.error('Error parsing banner YAML:', e);
+    return [];
+  }
+};
```

Comment:
use one of the array methods here, array.map would be good candiate.

Reply:
updated